packages/core/strapi/lib/commands/utils/commander.js

```diff
@@ -28,8 +28,13 @@ const parseInputList = (value) => {
 const promptEncryptionKey = async (thisCommand) => {
   const opts = thisCommand.opts();
 
+  if (!opts.encrypt && opts.key) {
+    console.error('Key may not be present unless encryption is used');
+    process.exit(1);
+  }
+
   // if encrypt is set but we have no key, prompt for it
-  if (opts.encrypt && !opts.key) {
+  if (opts.encrypt && !(opts.key && opts.key.length > 0)) {
```

Comment:
If a key is set but empty, do we want to prompt or throw an error saying the key should not empty?

Reply:
It should work like this, so the user can specify "I do for sure want to use a key, but I don't want to enter it directly on the command line"

Reply:
Changing my mind on this. If we allow --key to be empty, a user could accidentally attempt `yarn strapi export --key backup.tar` and encrypt their data with the key 'backup.tar' by mistake. We'll document in --encrypt that if it's true it will prompt, UNLESS key is set and then that will be used.